### ISSUE TITLE
Interactive mode: Error positions are always resolved relative to newest input

### ISSUE THREAD

**Issue by thechnet** [thechnet/une#5]
```
>>> a=function() 1/0
>>> a()
Error in file "<args>" at line 1,
In function {"<args>":1}:
a()
             ~~~
Zero division.
```

In the above example, the error should show the first input, `a=function() 1/0`, but it instead shows the newest, `a()`, making no sense.

There are two "quick" approaches I can come up with to solve this issue, but neither of them is good:
- Make `une_run()` aware of the larger buffer containing *all* the inputs, and let it modify the lexer state just before calling `une_error_display()`.
- Move `une_error_display()` out of `une_run()`. Unfortunately, that also requires moving the error container and the lexer state.

---

Issues like this one aren't unique (compare `une_run()`'s declaration now to before the addition of the interactive mode); they all stem from the (bad) decision to make `une_run()` as self-contained and locked-down as possible.

The only true solution I see is to forget `une_run()` and instead **simplify the steps that made up its body**.

**Comment by thechnet** [comment]
Addressed by 8fbde3d73565b692897aee8474fdaedd4351003b.

**Comment by thechnet** [comment]
Addressed by 8fbde3d73565b692897aee8474fdaedd4351003b.
